pandas_gbq/gbq.py

```diff
@@ -964,10 +967,16 @@ def read_gbq(
                 'Index column "{0}" does not exist in DataFrame.'.format(index_col)
             )
 
+    # Using columns as an alias for col_order, raising an error if both provided
+    if col_order and not columns:
+        columns = col_order
+    elif col_order and columns:
+        raise ValueError("Must specify either columns or col_order, not both")
+
     # Change the order of columns in the DataFrame based on provided list
-    if col_order is not None:
-        if sorted(col_order) == sorted(final_df.columns):
-            final_df = final_df[col_order]
+    if columns is not None:
+        if sorted(columns) == sorted(final_df.columns):
```

Comment:
This is a difference between how we're using `columns` in BigQuery DataFrames and pandas-gbq. We expect columns to be a subset of all columns in the table.

Please add a TODO to allow this to be a subset instead.

In a follow-up PR, please pass columns to download_table so that we can only download a subset of columns.

https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_list_rows has a selected_fields argument for this purpose.